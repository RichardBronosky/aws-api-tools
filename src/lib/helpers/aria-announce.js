const formatMsg = msg => {
  return '<div aria-live="assertive" aria-atomic="true">' + msg + '</div>'
}

export const taskDone = (fn, time, id) => {
  let timers = {}
  if (typeof fn == 'function') {
    if (timers[id]) clearTimeout(timers[id])
    timers[id] = setTimeout(fn, time)
  }
}

export const announce = (msg, form, ele, delay) => {
  if (typeof ele == 'object') {
    msg += ele.innerText.replace(/\n/g, ' ')
  }
  if (document) {
    document.getElementById('aria-announce').innerHTML = ''
    taskDone(
      () => {
        document.getElementById('aria-announce').innerHTML = formatMsg(msg)
      },
      delay || 500,
      'announceMessage'
    )
    if (form) {
      taskDone(
        () => {
          form.querySelectorAll('[aria-invalid="true"]')[0].focus()
        },
        1000,
        'focusElement'
      )
    }
  }
}
