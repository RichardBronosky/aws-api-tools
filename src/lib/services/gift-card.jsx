const axios = require('axios')

export const balanceInquiry = body =>
  axios.post(`${ process.env.GATSBY_GC_SERVICE_URL }/balanceInquiry/`, body).then(response => response.data)
