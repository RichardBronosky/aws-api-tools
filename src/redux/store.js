import React from 'react'
import { Provider } from 'react-redux'
import { createStore, compose, combineReducers } from 'redux'

import global from './modules/global'
import location from './modules/location'
import cart from './modules/cart'
import checkout from './modules/checkout'

const rootReducer = combineReducers({
  global,
  location,
  cart,
  checkout,
})

// Redux DevTools
const windowObject = typeof window !== 'undefined' && window
const reduxDevTools =
  process.env.NODE_ENV === 'development' && windowObject.devToolsExtension
    ? window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    : _ => _

const configureStore = initialState => createStore(rootReducer, initialState, compose(reduxDevTools))

export const store = configureStore()

// eslint-disable-next-line react/display-name,react/prop-types
export default ({ element }) => <Provider store={ store }>{ element }</Provider>
