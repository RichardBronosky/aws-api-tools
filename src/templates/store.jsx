import React from 'react'
import Helmet from 'react-helmet'
import { Map, Marker, InfoWindow, GoogleApiWrapper } from 'google-maps-react'
import Layout from '../components/layout'
import Breadcrumb from '../components/shared/breadcrumbs'
import RTGLink from '../components/shared/link'
import { decodeHtml, slugify, weekdays, months, getStandardTime } from '../lib/helpers/string-helper'
import { contentfulImage } from '../lib/helpers/contentful'
import { getStoreHours, getStoreName, getStoreInfoFromFeed } from '../lib/helpers/store-locator'
import { expandState } from '../lib/helpers/geo-location'
import { setupAnalytics } from '../lib/helpers/google-tag-manager'
import StoreInfo from '../components/store-locator/store-info'
import '../assets/css/pages/store-location.sass'

class Store extends React.Component {
  state = {
    showingInfoWindow: false,
    activeMarker: {},
    selectedMarker: {},
  }

  componentDidMount() {
    const store = this.props.pageContext.store
    setupAnalytics({ pageData: { type: 'store', title: getStoreName(store), path: window.location.pathname } })
  }

  createCrumb = (crumb, slugPath) => {
    return {
      altDesc: `View All ${ crumb } Products`,
      id: slugPath,
      slug: slugPath,
      text: crumb,
    }
  }

  generateStoreCrumbs = store => {
    let crumbs = []
    crumbs.push(this.createCrumb('Find a Showroom', '/stores'))
    if (store.state) {
      crumbs.push(
        this.createCrumb(
          `Find a Showroom - ${ expandState(store.state) }`,
          `/stores/${ slugify(expandState(store.state)) }`
        )
      )
    }
    crumbs.push(
      this.createCrumb(
        decodeHtml(`${ store.city }, ${ store.state }${ store.storeName ? ` (${ store.storeName })` : '' }`),
        window.location.pathname
      )
    )
    return {
      crumbs: crumbs,
    }
  }

  onLocationClick = (props, marker) => {
    const store = this.props.pageContext.store
    const storeInfoFromFeed = getStoreInfoFromFeed(store.storeNumber)
    let fullMarker = {
      ...props,
      title: store.address1,
      name: getStoreName(store),
      lat: store.location.lat,
      lng: store.location.lon,
      city: store.city,
      state: store.state,
      storeNumber: store.storeNumber,
      zip: store.zip,
      phone: storeInfoFromFeed.phone.replace('/', '-'),
      hours: getStoreHours(storeInfoFromFeed.store_hours),
    }
    this.setState({
      selectedMarker: fullMarker,
      activeMarker: marker,
      showingInfoWindow: true,
    })
  }

  onMapClicked = () => {
    if (this.state.showingInfoWindow) {
      this.setState({
        showingInfoWindow: false,
        activeMarker: null,
      })
    }
  }

  getTodayOpen = store => {
    const currentDay = weekdays[new Date().getDay()]
    const month = months[new Date().getMonth()].substr(0, 3)
    const day = new Date().getDate()
    const storeOpen = store.store_hours[`${ currentDay.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ currentDay.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ currentDay }, ${ month } ${ day }: ${
        storeOpen ? `Open from ${ getStandardTime(storeOpen) } to ${ getStandardTime(storeClose) }` : 'Closed'
      }`
    } else {
      ;`${ currentDay }, ${ month } ${ day }`
    }
  }

  getHours = (store, day) => {
    const storeOpen = store.store_hours[`${ day.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ day.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ getStandardTime(storeOpen) } - ${ getStandardTime(storeClose) }`
    }
  }

  render() {
    const { activeMarker, selectedMarker, showingInfoWindow } = this.state
    const store = this.props.pageContext.store
    const address = `${ store.address1 }${ store.address2 ? `, ${ store.address2 }` : '' }, ${ store.city }, ${ store.state } ${
      store.zip
    } `
    const storeInfoFromFeed = getStoreInfoFromFeed(store && store.storeNumber)
    if (store) {
      let storeTypeTitle = `${ store.city }, ${ store.state }`
      if (store.storeType === 'Showroom' || store.storeType === 'Express') {
        storeTypeTitle = storeTypeTitle + ' Furniture & Mattress Store'
      } else if (store.storeType === 'Kids') {
        storeTypeTitle = storeTypeTitle + ' Kids Furniture & Mattress Store'
      } else if (store.storeType === 'Outlet') {
        storeTypeTitle = storeTypeTitle + ' Affordable Furniture Outlet Store'
      } else if (store.storeType === 'Outdoor') {
        storeTypeTitle = storeTypeTitle + ' Patio Furniture: Wicker, Teak Outdoor Furniture Sets'
      }
      return (
        <Layout { ...this.props }>
          <Helmet title={ storeTypeTitle } />
          { store.seo && (
            <>
              { store.seo.pageTitle && <Helmet title={ store.seo.pageTitle } /> }
              { store.seo.canonical && (
                <Helmet
                  link={ [
                    {
                      rel: 'canonical',
                      href: store.seo.canonical,
                    },
                  ] }
                />
              ) }
              { store.seo.metaDescription && (
                <Helmet
                  meta={ [
                    {
                      name: 'description',
                      content: store.seo.metaDescription.metaDescription,
                    },
                  ] }
                />
              ) }
              { store.seo.metaRobots && (
                <Helmet
                  meta={ [
                    {
                      name: 'robots',
                      content: store.seo.metaRobots,
                    },
                  ] }
                />
              ) }
              { store.seo.jsonLdSchema && (
                <Helmet
                  script={ [
                    {
                      type: 'application/ld+json',
                      innerHTML: store.seo.jsonLdSchema.jsonLdSchema,
                    },
                  ] }
                />
              ) }
            </>
          ) }
          <section className="seo-page-heading full-width hide show">
            <div className="grid-container ">
              <div className="grid-x grid-margin-x">
                <div className="cell small-12">
                  <h1>
                    { store.seo && store.seo.pageHeading
                      ? store.seo.pageHeading
                      : `${ storeTypeTitle }${ store.storeName ? ` (${ store.storeName })` : '' }` }
                  </h1>
                </div>
              </div>
            </div>
          </section>
          <div className="stores-breadcrumb">
            <Breadcrumb data={ this.generateStoreCrumbs(store) } includeHeading={ true } />
          </div>
          <div className="store-location">
            <div className="loc-info">
              { address }
              <RTGLink
                data={ {
                  url: `tel:${ storeInfoFromFeed.phone.replace('/', '-') }`,
                  title: 'Call store',
                  category: 'store page',
                  action: 'Call click',
                  label: 'Call',
                } }
                className="phone-link"
              >
                { storeInfoFromFeed.phone.replace('/', '-') }
              </RTGLink>
              <span className="today">{ this.getTodayOpen(storeInfoFromFeed) }</span>
            </div>
            <div className="grid-x map-container">
              <Map
                initialCenter={ { lat: store.location.lat, lng: store.location.lon } }
                center={ { lat: store.location.lat, lng: store.location.lon } }
                className="map"
                google={ this.props.google }
                zoom={ 11 }
                onClick={ this.onMapClicked }
              >
                <Marker
                  position={ { lat: store.location.lat, lng: store.location.lon } }
                  title={ `${ store.city }${ store.storeName ? ` - ${ store.storeName }` : '' } ${ store.storeType }` }
                  icon={ {
                    url: `http://maps.google.com/mapfiles/markerA.png`,
                  } }
                  onClick={ this.onLocationClick }
                />
                <InfoWindow marker={ activeMarker } visible={ showingInfoWindow }>
                  <StoreInfo selectedMarker={ selectedMarker } />
                </InfoWindow>
              </Map>
              { store.storeImage && (
                <div className="store-image-container">
                  <img className="store-image" alt={ storeTypeTitle } src={ contentfulImage(store.storeImage.file.url) } />
                </div>
              ) }
              { storeInfoFromFeed.store_hours && (
                <div className="section-container">
                  <h2 className="section-heading">Store Hours:</h2>
                  { weekdays.map(day => (
                    <div className="hour" key={ day }>
                      { storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">{ this.getHours(storeInfoFromFeed, day) }</span>
                        </>
                      ) }
                      { !storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">Closed</span>
                        </>
                      ) }
                    </div>
                  )) }
                  { store.specialClosings && (
                    <span
                      className="special-closings"
                      dangerouslySetInnerHTML={ { __html: store.specialClosings.markdown.childMarkdownRemark.html } }
                    />
                  ) }
                </div>
              ) }
              { store.directions && (
                <div className="section-container">
                  <h2 className="section-heading">Store Location / Address:</h2>
                  <span>{ address }</span>
                  <br />
                  <span
                    className="directions"
                    dangerouslySetInnerHTML={ { __html: store.directions.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.neighborhoodsNearby && store.neighborhoodsNearby.length > 0 && (
                <div className="section-container">
                  <h2 className="section-heading">Neighborhoods Nearby:</h2>
                  <ul>
                    { store.neighborhoodsNearby.map((hood, index) => (
                      <li key={ index } className="hood">
                        { hood }
                      </li>
                    )) }
                  </ul>
                </div>
              ) }
              { store.description && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">Store Description:</h2>
                  <span
                    className="description"
                    dangerouslySetInnerHTML={ { __html: store.description.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.faqs && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">FAQs:</h2>
                  <span
                    className="faqs"
                    dangerouslySetInnerHTML={ { __html: store.faqs.markdown.childMarkdownRemark.html } }
                  />
                </div>
              ) }
            </div>
          </div>
        </Layout>
      )
    } else {
      return null
    }
  }
}

export default GoogleApiWrapper({
  apiKey: `${ process.env.GATSBY_GOOGLE_BROWSER_API_KEY }`,
})(Store)
