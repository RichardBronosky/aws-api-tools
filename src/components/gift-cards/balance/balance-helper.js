import { balanceInquiry } from '@services/gift-card'
import { announce } from '@helpers/aria-announce'

export const submitGC = async (e, cardNumber, pin, setError, setLoading, setBalance) => {
  e.preventDefault()
  setLoading(true)
  const valid = validateCardNumbers(cardNumber, pin)
  if (valid) {
    let data
    try {
      data = await balanceInquiry({
        cardNumber,
        pin,
      })
    } catch (err) {
      const techDetails = err && err.response && err.response.data && err.response.data.error && err.response.data.error
      if (techDetails) {
        if (techDetails.includes('Account closed')) {
          submitErr(setError, '*This account has been closed.')
        }
      } else {
        submitErr(setError)
      }
    }
    if (data && data.giftCardBalance) {
      setBalance(data.giftCardBalance)
      setError(false)
      announce(`Form Submitted: Your card has a balance of ${ data.giftCardBalance }`)
    } else {
      submitErr(setError, '*The card number or pin entered are invalid. Please verify and try again.')
    }
  } else {
    submitErr(setError, '*The card number or pin entered are invalid. Please verify and try again.')
  }
  setLoading(false)
}

export const submitErr = (setError, error) => {
  let err = error || '*We were unable to find the card entered. Please verify the card number and pin and try again.'
  announce('Form Submitted: ' + err)
  setError(err)
}

export const validateCardNumbers = (number, pin) => {
  var reNum = /^[0-9]{16}/
  var rePin = /^[0-9]{8}/
  return reNum.test(String(number).toLowerCase()) && rePin.test(String(pin).toLowerCase())
}
