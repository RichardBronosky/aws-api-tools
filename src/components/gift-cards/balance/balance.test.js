import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { store } from '@redux/store'
import GiftCardBalance from './index'
import * as balanceHelper from './balance-helper'
import * as gcService from '@services/gift-card'
import * as announceHelper from '@helpers/aria-announce'

describe('GiftCardBalance', () => {
  const e = { preventDefault: jest.fn() }

  it('renders correctly when desktop', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <GiftCardBalance isMobile={ false } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when mobile', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <GiftCardBalance isMobile={ true } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('simulates onChange for gift card number', () => {
    const tree = renderer.create(
      <Provider store={ store }>
        <GiftCardBalance isMobile={ false } />
      </Provider>
    )
    renderer.act(() =>
      tree.root.findByProps({ name: 'gift_card_number' }).props.onChange({ target: { value: 'test' } })
    )
  })

  it('simulates onChange for gift card pin', () => {
    const tree = renderer.create(
      <Provider store={ store }>
        <GiftCardBalance isMobile={ false } />
      </Provider>
    )
    renderer.act(() => tree.root.findByProps({ name: 'gift_card_pin' }).props.onChange({ target: { value: 'test' } }))
  })

  it('submitGC errors when invalid card number', () => {
    announceHelper.announce = jest.fn()
    const setError = jest.fn()
    const setLoading = jest.fn()
    const setBalance = jest.fn()
    balanceHelper.submitGC(e, '1234', '1234', setError, setLoading, setBalance)
    expect(setError).toHaveBeenCalledWith('*The card number or pin entered are invalid. Please verify and try again.')
    expect(announceHelper.announce).toHaveBeenCalled()
    expect(setLoading).toHaveBeenCalled()
  })

  it('submitGC errors when when account is closed', () => {
    gcService.balanceInquiry = jest.fn(() =>
      Promise.reject({
        response: {
          data: {
            error: 'Account closed',
          },
        },
      })
    )
    announceHelper.announce = jest.fn()
    const setError = jest.fn()
    const setLoading = jest.fn()
    const setBalance = jest.fn()
    balanceHelper.submitGC(e, '1234123412341234', '12341234', setError, setLoading, setBalance)
    expect(setLoading).toHaveBeenCalled()
  })

  it('submitGC errors when bad api response', () => {
    gcService.balanceInquiry = jest.fn(() => Promise.reject())
    announceHelper.announce = jest.fn()
    const setError = jest.fn()
    const setLoading = jest.fn()
    const setBalance = jest.fn()
    balanceHelper.submitGC(e, '1234123412341234', '12341234', setError, setLoading, setBalance)
    expect(setLoading).toHaveBeenCalled()
  })

  it('submitGC sets balance when there is a gift card balance', () => {
    gcService.balanceInquiry = jest.fn(() =>
      Promise.resolve({
        giftCardBalance: '500',
      })
    )
    announceHelper.announce = jest.fn()
    const setError = jest.fn()
    const setLoading = jest.fn()
    const setBalance = jest.fn()
    balanceHelper.submitGC(e, '1234123412341234', '12341234', setError, setLoading, setBalance)
    expect(setLoading).toHaveBeenCalled()
  })

  it('simulates onClick for form submission', () => {
    balanceHelper.submitGC = jest.fn()
    const tree = renderer.create(
      <Provider store={ store }>
        <GiftCardBalance isMobile={ false } />
      </Provider>
    )
    tree.root.findByProps({ className: 'primary' }).props.onClick()
  })
})
