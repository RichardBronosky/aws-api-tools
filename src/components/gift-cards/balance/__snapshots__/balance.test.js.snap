// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftCardBalance renders correctly when desktop 1`] = `
<div
  className="gc-balance cell card"
>
  <div
    className="grid-x"
  >
    <div
      className="cell small-12"
    >
      <h1>
        Check Gift Card Balance
      </h1>
    </div>
    <div
      className="cell small-12 grid-x"
    >
      <div
        className="cell grid-x small-12 large-6 numbers-image"
      >
        <img
          alt="Gift Card Number Locations"
          className="small-12"
          src="undefinedgiftcard-diagram.jpg"
        />
        <div
          className="redeem-message"
        >
          <a
            aria-describedby="purchase-gift-card"
            className="small-12 period"
            gtm-action="click"
            gtm-category="gift card"
            gtm-label="purchase gift card"
            gtm-value={null}
            href="/gift-cards/purchase"
            onBlur={null}
            onClick={[Function]}
            onFocus={null}
            onKeyDownCapture={null}
          >
            Purchase Gift Card 
          </a>
          <span>
            Redeem your Gift Card
            <a
              aria-describedby="redeem-gift-card-online"
              className="small-12"
              gtm-action="click"
              gtm-category="gift card"
              gtm-label="online"
              gtm-value={null}
              href="/"
              onBlur={null}
              onClick={[Function]}
              onFocus={null}
              onKeyDownCapture={null}
            >
               online 
            </a>
            or in your nearest
            <a
              aria-describedby="redeem-gift-card-in-showroom"
              className="small-12 period"
              gtm-action="click"
              gtm-category="gift card"
              gtm-label="showroom"
              gtm-value={null}
              href="/stores"
              onBlur={null}
              onClick={[Function]}
              onFocus={null}
              onKeyDownCapture={null}
            >
               showroom
            </a>
            .
          </span>
        </div>
      </div>
      <div
        className="grid-x small-12 large-6 form-area"
      >
        <div
          className="small-12"
        >
          <p>
            Fill in the fields below to retrieve your gift card balance.
          </p>
          <p
            className="asterisk"
          >
            *Asterisks indicate required fields in the form below
          </p>
        </div>
        <form
          className="grid-x cell small-8"
        >
          <label
            className="cell small-12"
          >
            16-digit Card Number
          </label>
          <input
            className="cell small-12"
            id="gift_card_number"
            maxLength={16}
            name="gift_card_number"
            onChange={[Function]}
            type="text"
            value=""
          />
          <label
            className="cell small-12"
          >
            8-digit PIN Number
          </label>
          <input
            className="cell small-12"
            id="gift_card_pin"
            maxLength={8}
            name="gift_card_pin"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="primary"
            gtm-action="gift card balance check"
            gtm-category="gift-card"
            gtm-label="gift-card-balance"
            onClick={[Function]}
          >
            Check Card Balance
          </button>
        </form>
      </div>
    </div>
    <div
      className="cell accordians"
    >
      <div
        className="gc-accordions"
      >
        <div
          className="contentful-accordion"
        >
          <div
            aria-controls="gf-faqs"
            aria-expanded={false}
            className="contentful-accordion-heading"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <span>
              Gift Card FAQs
            </span>
            <i
              className=""
            />
          </div>
          <div
            className="contentful-accordion-content"
          >
            <div
              className="gc-faq-wrapper"
            >
              <span
                className="gc-faq-title"
              >
                Can I buy Rooms To Go gift cards online?
              </span>
              <span>
                Yes, you may purchase Rooms To Go gift cards online at roomstogo.com.
              </span>
            </div>
            <div
              className="gc-faq-wrapper"
            >
              <span
                className="gc-faq-title"
              >
                How do I redeem a Rooms To Go gift card online?
              </span>
              <span>
                Shop roomstogo.com, choose merchandise you wish to buy and add it to your shopping cart. At Checkout, a space is provided to enter the gift card number, which is located on the back of the gift card, as well as the PIN number, also provided on the back.
              </span>
            </div>
            <div
              className="gc-faq-wrapper"
            >
              <span
                className="gc-faq-title"
              >
                Can I redeem a Rooms To Go gift card in a Rooms To Go showroom?
              </span>
              <span>
                Yes, your gift card may be redeemed in any Rooms To Go or Rooms To Go Kids showroom in the Contiguous U.S. Sorry, gift cards cannot be redeemed in showrooms outside the United States at this time.
              </span>
            </div>
            <div
              className="gc-faq-wrapper"
            >
              <span
                className="gc-faq-title"
              >
                What are the shipping charges for Rooms To Go gift cards?
              </span>
              <span>
                Your gift card will be sent free of charge via certified mail. Please be aware that gift cards will only be mailed to the purchaser's billing address, not your intended recipient. This is for security purposes.
              </span>
            </div>
          </div>
        </div>
        <div
          className="contentful-accordion"
        >
          <div
            aria-controls="gf-terms"
            aria-expanded={false}
            className="contentful-accordion-heading"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <span>
              Terms and conditions
            </span>
            <i
              className=""
            />
          </div>
          <div
            className="contentful-accordion-content"
          >
            <div>
              <p>
                The Rooms To Go gift card is easy to carry and easy to use. Once activated, this card is redeemable for merchandise at any Rooms To Go or Rooms To Go Kids showroom, or online at roomstogo.com. Please note that the pin number from the gift card is required to redeem online. Checking the gift card balance is simple, by calling 1-888-228-0144 or by
                <a
                  gtm-action="click"
                  gtm-category="gift-card"
                  gtm-label="gift card balance"
                  gtm-value={null}
                  href="/gift-cards/balance"
                  onBlur={null}
                  onClick={[Function]}
                  onFocus={null}
                  onKeyDownCapture={null}
                  title="Rooms To Go Gift Card Balance"
                >
                   clicking here
                </a>
                . The Rooms To Go gift card will be mailed to you at no charge. It is our policy to mail the card to the purchaser's billing address. This card has no cash value and is not redeemable for cash, nor will it be replaced if lost or stolen. Gift cards may not be applied towards prior purchases or account balances or to purchase additional Gift cards. Gift cards are not returnable or refundable. Any balance remains on the Gift card until it is spent. The Rooms To Go gift card is an ideal gift choice for weddings, graduations, anniversaries and so many other special occasions.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GiftCardBalance renders correctly when mobile 1`] = `
<div
  className="gc-balance cell card"
>
  <div
    className="grid-x"
  >
    <div
      className="cell small-12"
    >
      <h1>
        Check Gift Card Balance
      </h1>
    </div>
    <div
      className="cell small-12 grid-x"
    >
      <div
        className="cell grid-x small-12 large-6 numbers-image"
      >
        <img
          alt="Gift Card Number Locations"
          className="small-12"
          src="undefinedgiftcard-diagram.jpg"
        />
        <div
          className="redeem-message"
        >
          <a
            aria-describedby="purchase-gift-card"
            className="small-12 period"
            gtm-action="click"
            gtm-category="gift card"
            gtm-label="purchase gift card"
            gtm-value={null}
            href="/gift-cards/purchase"
            onBlur={null}
            onClick={[Function]}
            onFocus={null}
            onKeyDownCapture={null}
          >
            Purchase Gift Card 
          </a>
          <span>
            Redeem your Gift Card
            <a
              aria-describedby="redeem-gift-card-online"
              className="small-12"
              gtm-action="click"
              gtm-category="gift card"
              gtm-label="online"
              gtm-value={null}
              href="/"
              onBlur={null}
              onClick={[Function]}
              onFocus={null}
              onKeyDownCapture={null}
            >
               online 
            </a>
            or in your nearest
            <a
              aria-describedby="redeem-gift-card-in-showroom"
              className="small-12 period"
              gtm-action="click"
              gtm-category="gift card"
              gtm-label="showroom"
              gtm-value={null}
              href="/stores"
              onBlur={null}
              onClick={[Function]}
              onFocus={null}
              onKeyDownCapture={null}
            >
               showroom
            </a>
            .
          </span>
        </div>
      </div>
      <div
        className="grid-x small-12 large-6 form-area"
      >
        <div
          className="small-12"
        >
          <p>
            Fill in the fields below to retrieve your gift card balance.
          </p>
          <p
            className="asterisk"
          >
            *Asterisks indicate required fields in the form below
          </p>
        </div>
        <form
          className="grid-x cell small-8"
        >
          <label
            className="cell small-12"
          >
            16-digit Card Number
          </label>
          <input
            className="cell small-12"
            id="gift_card_number"
            maxLength={16}
            name="gift_card_number"
            onChange={[Function]}
            type="text"
            value=""
          />
          <label
            className="cell small-12"
          >
            8-digit PIN Number
          </label>
          <input
            className="cell small-12"
            id="gift_card_pin"
            maxLength={8}
            name="gift_card_pin"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="primary"
            gtm-action="gift card balance check"
            gtm-category="gift-card"
            gtm-label="gift-card-balance"
            onClick={[Function]}
          >
            Check Card Balance
          </button>
        </form>
      </div>
    </div>
    <div
      className="cell accordians"
    >
      <div
        className="gc-accordions"
      >
        <div
          className="contentful-accordion"
        >
          <div
            aria-controls="gf-faqs"
            aria-expanded={false}
            className="contentful-accordion-heading"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <span>
              Gift Card FAQs
            </span>
            <i
              className=""
            />
          </div>
          <div
            className="contentful-accordion-content"
          >
            <div
              className="gc-faq-wrapper"
            >
              <span
                className="gc-faq-title"
              >
                Can I buy Rooms To Go gift cards online?
              </span>
              <span>
                Yes, you may purchase Rooms To Go gift cards online at roomstogo.com.
              </span>
            </div>
            <div
              className="gc-faq-wrapper"
            >
              <span
                className="gc-faq-title"
              >
                How do I redeem a Rooms To Go gift card online?
              </span>
              <span>
                Shop roomstogo.com, choose merchandise you wish to buy and add it to your shopping cart. At Checkout, a space is provided to enter the gift card number, which is located on the back of the gift card, as well as the PIN number, also provided on the back.
              </span>
            </div>
            <div
              className="gc-faq-wrapper"
            >
              <span
                className="gc-faq-title"
              >
                Can I redeem a Rooms To Go gift card in a Rooms To Go showroom?
              </span>
              <span>
                Yes, your gift card may be redeemed in any Rooms To Go or Rooms To Go Kids showroom in the Contiguous U.S. Sorry, gift cards cannot be redeemed in showrooms outside the United States at this time.
              </span>
            </div>
            <div
              className="gc-faq-wrapper"
            >
              <span
                className="gc-faq-title"
              >
                What are the shipping charges for Rooms To Go gift cards?
              </span>
              <span>
                Your gift card will be sent free of charge via certified mail. Please be aware that gift cards will only be mailed to the purchaser's billing address, not your intended recipient. This is for security purposes.
              </span>
            </div>
          </div>
        </div>
        <div
          className="contentful-accordion"
        >
          <div
            aria-controls="gf-terms"
            aria-expanded={false}
            className="contentful-accordion-heading"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <span>
              Terms and conditions
            </span>
            <i
              className=""
            />
          </div>
          <div
            className="contentful-accordion-content"
          >
            <div>
              <p>
                The Rooms To Go gift card is easy to carry and easy to use. Once activated, this card is redeemable for merchandise at any Rooms To Go or Rooms To Go Kids showroom, or online at roomstogo.com. Please note that the pin number from the gift card is required to redeem online. Checking the gift card balance is simple, by calling 1-888-228-0144 or by
                <a
                  gtm-action="click"
                  gtm-category="gift-card"
                  gtm-label="gift card balance"
                  gtm-value={null}
                  href="/gift-cards/balance"
                  onBlur={null}
                  onClick={[Function]}
                  onFocus={null}
                  onKeyDownCapture={null}
                  title="Rooms To Go Gift Card Balance"
                >
                   clicking here
                </a>
                . The Rooms To Go gift card will be mailed to you at no charge. It is our policy to mail the card to the purchaser's billing address. This card has no cash value and is not redeemable for cash, nor will it be replaced if lost or stolen. Gift cards may not be applied towards prior purchases or account balances or to purchase additional Gift cards. Gift cards are not returnable or refundable. Any balance remains on the Gift card until it is spent. The Rooms To Go gift card is an ideal gift choice for weddings, graduations, anniversaries and so many other special occasions.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
